### ISSUE TITLE
Cython 3 compilation

### ISSUE THREAD

**Issue by Vutshi** [QuSpin/QuSpin#642]
**Describe the bug**
Error compiling Cython file

**To Reproduce**
Compile on macOS with cython 3.0 to get a lot of error messages:

```
Error compiling Cython file:
------------------------------------------------------------
...
                    raise TypeError("attempting to use real type for complex elements.")

                with nogil:
                    for i in range(Ns):
                        c[i] *= sign[i]
                        indices[i] = <npy_intp>rep_position(Ns_full,basis_pcon,basis[i])
                                                           ^
------------------------------------------------------------

quspin/basis/basis_general/_basis_general_core/source/general_basis_core.pyx:128:59: Unable to deduce type parameters for K (const K, I *, const I) noexcept nogil given (npy_intp, uint256_t *, uint256_t &)
```

**Desktop (please complete the following information):**
 - OS: macOS 13.5.2
 - Python 3.11.5
 - Cython 3.0.0

**Comment by mgbukov** [comment]
downgrade cython to 0.29 (see version prereqs online) and it should be fine.

Marin

**Comment by mgbukov** [comment]
There are many plans but very little developer time on our side, so if you
wanna go ahead and fix it we'd be happy to consider a commit into the most
recent development branch.

For some reason cython decided to break backward compatibility...

Marin

**Comment by mgbukov** [comment]
this should be fixed with quspin 1.0.0
